src/bid/bid.rs

```diff
@@ -24,7 +30,7 @@ pub struct Bid {
     // i (One time identity of the prover)
     pub(crate) prover_id: Option<Scalar>,
     // q (Score of the bid)
-    pub(crate) score: Option<Scalar>,
+    pub(crate) score: Option<Score>,
```

Comment:
These two should not be Options, since all the code assumes they are already computed, and they're being computed in the new function. Just turn them into normal Scalars.

Also, is the Default implementation really needed?

Reply:
I agree on the `Option` part. I refactored the lib to follow this new logic.

Regarding the `Default` impl. Is needed for `Score` but not for the `Bid` that we are aware of at least now. So I will remove it.